compiler/Cargo.toml

```diff
@@ -1 +1,14 @@
-TBD
+[package]
+name = "compiler"
+version = "0.1.0"
+
+[lib]
+name = "compiler"
+path = "src/lib.rs"
+
+[[bin]]
+name = "testbin"
```

Comment:
Do we need this crate to be a separate program?

Reply:
> Do we need this crate to be a separate program?

Of course, no - compiler is a lib in fact. When we run "cargo build", we get lib. And "cargo run" allows to run main function in "src/bin/compiler_main.rs" (I have read about this possibility, when I prepared my presentation about crates and modules and tried to explore it).
This option gives us two benefits:
1. Possibility of direct Compiler debugging - we have "main" function in library.
2. Use case (example) of the Compiler usage by backend "src/bin/compiler_main.rs".

So, we can remove it anytime.

Are there (or will be in the future) some critical  issues with this approach?